File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> seen = new HashMap<>();

        for (int i = 0; i < nums.length; i++){
            int num = nums[i];
            int numToLookup = target - num;
            
            if(seen.containsKey(numToLookup)){
                return new int[]{i, seen.get(numToLookup)};
            }

            seen.put(num, i);
        }

        return new int[]{-1, -1};
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        for i, x in enumerate(nums):
            complement = target - x

            for j in range(i + 1, len(nums)):
                if nums[j] == complement:
                    return [i, j]

        return [-1, -1]
